mne/channels/channels.py

```diff
@@ -987,6 +989,118 @@ def add_channels(self, add_list, force_update_info=False):
             assert all(len(r) == self.info['nchan'] for r in self._read_picks)
         return self
 
+    def combine_channels(self, group_by, method='mean'):
+        """Combine channels based on regions of interest (ROIs).
+
+        Parameters
+        ----------
+        group_by : dict
+            Specifies which channels are aggregated into a single channel, with
+            aggregation method determined by the ``method`` parameter. One new
+            pseudo-channel is made per dict entry; the dict values must be
+            lists of picks (integer indices of ``ch_names``). For example::
+
+                group_by=dict(Left_ROI=[1, 2, 3, 4], Right_ROI=[5, 6, 7, 8])
+
+            Note that within a dict entry all channels must have the same type.
+        %(method)s
+            If callable, the callable must accept one positional input (data of
+            shape ``(n_channels, n_times)``, ``(n_epochs, n_channels,
+            n_times)``, or ``(n_evokeds, n_channels, n_time)``) and return an
+            :class:`array <numpy.ndarray>` of shape ``(n_times)``, ``(n_epochs,
+            n_times)``, or ``(n_evokeds, n_times)``. For example with an
+            instance of Raw::
+
+                method = lambda data: np.median(data, axis=0)
+
+            Defaults to ``mean``.
+
+        Returns
+        -------
+        inst : instance of Raw, Epochs, or Evoked
+            A new instance.
+        """
+        from ..io import BaseRaw, RawArray
+        from ..epochs import BaseEpochs
+        from ..evoked import Evoked
+        from .. import EpochsArray, EvokedArray
```

Comment:
a quick `git grep` shows me that `_BaseTFR` inherits from `UpdateChannelsMixin`, so this will need to support TFR objects too, I think.